### DISCUSSION TITLE
[General]: Clicking on input elements after dragging

### DISCUSSION THREAD

**Discussion by Ra0R** [davidjerleke/embla-carousel#739] [General]
### Summary

Currently when dragging (with the mouse) with options:
`	const [emblaRef, emblaApi] = useEmblaCarousel({
		loop: false,
		containScroll: 'trimSnaps',
		skipSnaps: true
	});
`

The carousel takes a long time to settle and elements to become clickable again. Is there an option to enable a sampling rate or something similar?

Currently when listening to event scroll:
`		emblaApi.on('scroll', () => {
			console.log(emblaApi.scrollProgress())
		})`

The scroll value is updated a lot with of very small values and takes a long time to settle:
![image](https://github.com/davidjerleke/embla-carousel/assets/22443382/d2077735-770b-41b1-8ef8-92c30143d640)

Is it possible to define some minimum, e.g. delta = 0.01, such that value changes below that are ignored?

### If applicable, which variants of Embla Carousel are relevant to this conversation?

- [ ] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by davidjerleke** [comment]
@Ra0R Embla doesn't prevent any clicks when the carousel already has been dragged and is in motion. It only prevents clicks when the pointer is pressed and it registers a drag interaction rather than a static click - Which can be configured with the [`dragThreshold`](https://www.embla-carousel.com/api/options/#dragthreshold) option.

I'm not sure I understand what the problem is. Could you elaborate?

Best,
David

**Reply by Ra0R** [reply]
If you go to www.***.ch in the section "Das sagen unsere Kund:innen" the Embla library is used. After draging and letting the scroll end (without clicking again) the "Weiterlesen" click handler does not seem to fire. 
Clicking on anywhere else  (e.g. in the text above the "Weiterlesen" button) and then clicking on "Weiterlesen" seems to fix the problem.

**Reply by Ra0R** [reply]
I found a solution, I am not sure if this is a problem in Embla:
The click handler was on a `<input type="button" onClick={...}>` now that I switched the element with a `<span>` the problem seems to be resolved.

**Reply by davidjerleke** [reply]
@Ra0R [input elements](https://github.com/davidjerleke/embla-carousel/blob/master/packages/embla-carousel/src/components/DragHandler.ts#L61) won't trigger the Embla drag behaviour because they're focus nodes. I'm not sure if the problem you're seeing is the default browser behaviour or if it's something else that's causing this. If you can prove that it's Embla, I'll look into it.

Thanks!
David

**Reply by Ra0R** [reply]
@davidjerleke I created a CodeSandbox to demonstrate the problem:

[OnClick Handler Sandbox](https://codesandbox.io/p/sandbox/embla-carousel-generator-react-forked-58xddd?file=%2Fsrc%2Fjs%2FHeader.tsx%3A5%2C75&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clsog42fl00062a6clhe0rmhh%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clsog42fl00022a6c7781nbqj%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clsog42fl00032a6clh7ohy40%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clsog42fl00052a6cscq1stzi%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clsog42fl00022a6c7781nbqj%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsog42fl00012a6cahn0i7bu%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fjs%252Findex.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clsog57ka00022a6c6vxbcksj%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A5%252C%2522startColumn%2522%253A75%252C%2522endLineNumber%2522%253A5%252C%2522endColumn%2522%253A75%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252Fjs%252FHeader.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clsog42fl00022a6c7781nbqj%2522%252C%2522activeTabId%2522%253A%2522clsog57ka00022a6c6vxbcksj%2522%257D%252C%2522clsog42fl00052a6cscq1stzi%2522%253A%257B%2522id%2522%253A%2522clsog42fl00052a6cscq1stzi%2522%252C%2522activeTabId%2522%253A%2522clsog42fl00042a6cn7jihiu4%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsog42fl00042a6cn7jihiu4%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%252C%257B%2522type%2522%253A%2522ASSISTANT_CHAT%2522%252C%2522chatId%2522%253A%2522clsog90fy00cs2a6cdi1m0i0k%2522%252C%2522id%2522%253A%2522clsog90g000cu2a6c4svpocuz%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522clsog42fl00032a6clh7ohy40%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clsog42fl00032a6clh7ohy40%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D
)
Essentially the onClick handler of the button does not fire (unless you click the image above to kind of "halt" the carousel).

**Reply by davidjerleke** [reply]
Thanks @Ra0R. On devices where the primary pointer is a mouse (often desktop) or touch (often tablet and mobile phones)?

Which browser are you using? Have you tried different browsers?

**Reply by Ra0R** [reply]
@davidjerleke 
I tried Chrome and Firefox (mouse) and the issue is present in both. Haven't tested any touch devices.

**Reply by davidjerleke** [reply]
@Ra0R would you mind sharing a screen recording of the problem? I’m not sure if I understand what you mean.

**Reply by Ra0R** [reply]
You can see the button works only after I click the image above (click not really visible):

https://github.com/davidjerleke/embla-carousel/assets/22443382/3f7c1795-d49a-4fbc-9b36-b280ba28d140

**Reply by davidjerleke** [reply]
@Ra0R thanks! That was much more clear 👍🏻! I’ll investigate when possible.

**Reply by davidjerleke** [reply]
@Ra0R I can confirm that this is a bug. Nice catch 👍. I think we need to reset the `preventClick` variable here:

https://github.com/davidjerleke/embla-carousel/blob/b04f0afaf28fc8441e287db07922ab60db618c0c/packages/embla-carousel/src/components/DragHandler.ts#L189-L194

...like so: 
```ts
function click(evt: MouseEvent): void {
  if (preventClick) {
    evt.stopPropagation()
    evt.preventDefault()
  }
  preventClick = false // <-- here
} 
```

Haven't tested it yet but that seems to be the problem.

Best,
David

**Reply by davidjerleke** [reply]
Hi @Ra0R,

I would very much appreciate if you could test this [CodeSandbox](https://codesandbox.io/p/sandbox/embla-carousel-generator-react-forked-6mrvqr) and see if you can reproduce the bug there.

Best,
David

**Reply by davidjerleke** [reply]
@Ra0R I noticed that the [sandbox](https://codesandbox.io/p/sandbox/embla-carousel-generator-react-forked-6mrvqr) was private. Made it public now. Sorry about that!

**Reply by Ra0R** [reply]
@davidjerleke Thank you, it looks like the  issue is solved in the sandbox you sent!

**Reply by davidjerleke** [reply]
Thanks for confirming and providing useful information like the screen recording @Ra0R 👍! I will create a bug fix for this.

Best,
David

**Reply by davidjerleke** [reply]
@Ra0R this bug is now tracked here:

- https://github.com/davidjerleke/embla-carousel/issues/771

Closing this. Thank you 👍!
